packages/x-date-pickers/src/CalendarPicker/DayPicker.tsx

```diff
@@ -197,11 +205,14 @@ export function DayPicker<TDate>(props: DayPickerProps<TDate>) {
           <PickersCalendarWeekContainer
             data-mui-test="pickers-calendar"
             ref={slideNodeRef}
-            role="grid"
+            role="rowgroup"
```

Comment:
Please add `role="presentation"` to `PickersCalendarSlideTransition` to remove it from the accessibility tree.

Before: 

![image](https://user-images.githubusercontent.com/42154031/185398641-ca742410-fa2e-4fce-913e-1aa22c6bd906.png)

After:

![image](https://user-images.githubusercontent.com/42154031/185398719-73ef7b80-1154-4fce-a9d3-5e49de70d9ee.png)